2024-06-17 00:22:47,833 - INFO - Thông tin đầu vào:
2024-06-17 00:22:47,834 - INFO - file_name: datasets/training_user_based.xlsx
2024-06-17 00:22:47,834 - INFO - min_quota: 0
2024-06-17 00:22:47,834 - INFO - k: 2
2024-06-17 00:22:47,834 - INFO - ----------------------------------------------
2024-06-17 00:22:47,834 - INFO - Kết quả thực thi:
2024-06-17 00:22:47,918 - INFO - Các cặp (user, item) bị trùng lặp:
2024-06-17 00:22:47,918 - INFO -       user  item  rating
27      17    76    15.0
28      17    76   310.0
178     17   227     5.0
179     17   227    11.0
897     29    49    12.0
898     29    49    88.0
1077    30    49   302.0
1078    30    49   214.0
1329    33   121    23.0
1330    33   121    42.0
1437    34    49     4.0
1438    34    49     5.0
1439    34    49    17.0
1642    35    73     5.0
1643    35    73    18.0
2000    37    72     5.0
2001    37    72     5.0
3432    45    71    12.0
3433    45    71   276.0
3435    45    73    19.0
3436    45    73    32.0
3437    45    73     5.0
4330    50    71     8.0
4331    50    71   174.0
4497    51    58     8.0
4498    51    58  1120.0
4508    51    68    20.0
4509    51    68     8.0
4696    52    76    38.0
4697    52    76    10.0
4698    52    76    14.0
4874    53    73   186.0
4875    53    73   250.0
5032    54    51    12.0
5033    54    51     5.0
5390    56    50    88.0
5391    56    50     4.0
5392    56    50   548.0
5414    56    72    16.0
5415    56    72    30.0
5771    58    70     4.0
5772    58    70     8.0
2024-06-17 00:22:47,927 - ERROR - Có lỗi xảy ra: [Errno 13] Permission denied: 'result/1_original_matrix.xlsx'
Traceback (most recent call last):
  File "/Applications/XAMPP/xamppfiles/htdocs/GOATshop/admin/pages/thoi-gian-theo-doi/main.py", line 195, in main
    data_tuple = load_data(file_name, folder_name, min_quota)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Applications/XAMPP/xamppfiles/htdocs/GOATshop/admin/pages/thoi-gian-theo-doi/main.py", line 63, in load_data
    df_pivot.to_excel(os.path.join(folder_name, '1_original_matrix.xlsx'), index=False)
  File "/opt/anaconda3/envs/python=3.9/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/python=3.9/lib/python3.12/site-packages/pandas/core/generic.py", line 2417, in to_excel
    formatter.write(
  File "/opt/anaconda3/envs/python=3.9/lib/python3.12/site-packages/pandas/io/formats/excel.py", line 943, in write
    writer = ExcelWriter(
             ^^^^^^^^^^^^
  File "/opt/anaconda3/envs/python=3.9/lib/python3.12/site-packages/pandas/io/excel/_openpyxl.py", line 61, in __init__
    super().__init__(
  File "/opt/anaconda3/envs/python=3.9/lib/python3.12/site-packages/pandas/io/excel/_base.py", line 1246, in __init__
    self._handles = get_handle(
                    ^^^^^^^^^^^
  File "/opt/anaconda3/envs/python=3.9/lib/python3.12/site-packages/pandas/io/common.py", line 882, in get_handle
    handle = open(handle, ioargs.mode)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: 'result/1_original_matrix.xlsx'
